File: app/src/androidTest/java/tourguide/tourguidedemo/BasicActivityCrashTest.java
package tourguide.tourguidedemo;

        import android.test.ActivityInstrumentationTestCase2;

        import tourguide.tourguide.FrameLayoutWithHole;

/**
 * Created by tanjunrong on 8/6/15.
 */
public class BasicActivityCrashTest extends ActivityInstrumentationTestCase2<BasicActivity> {
    BasicActivity mActivity;
    FrameLayoutWithHole mOverlay;
    public BasicActivityCrashTest() {
        super(BasicActivity.class);
    }
    @Override
    protected void setUp() throws Exception {
        super.setUp();

        // Starts the activity under test using the default Intent with:
        // action = {@link Intent#ACTION_MAIN}
        // flags = {@link Intent#FLAG_ACTIVITY_NEW_TASK}
        // All other fields are null or empty.
        mActivity = getActivity();
        mOverlay = mActivity.mTutorialHandler.getOverlay();
    }

    public void testNoCrash(){
        mOverlay.performClick();
        assertTrue("There is no crash :)", true);
    }
}

File: app/src/androidTest/java/tourguide/tourguidedemo/BasicActivityCrashTest.kt
package tourguide.tourguidedemo

import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.action.ViewActions.click
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.support.test.rule.ActivityTestRule
import org.hamcrest.Matchers.allOf
import org.junit.Assert
import org.junit.Rule
import org.junit.Test

class BasicActivityCrashTest {
    @Rule @JvmField
    var rule: ActivityTestRule<BasicActivity> = ActivityTestRule(BasicActivity::class.java)

    @Test
    fun testNoCrash() {
        onView(allOf(withId(R.id.button1))).perform(click())
        Assert.assertTrue("There is no crash :)", true)
    }
}
